enhancements/builds/disable-builder-sa.md

```diff
@@ -0,0 +1,362 @@
+last-updated: 2024-02-06
+tracking-link:
+  - "https://issues.redhat.com/browse/BUILD-730"
+see-also:
+  - "https://issues.redhat.com/browse/RHDP-732"
+  - "https://issues.redhat.com/browse/OCPSTRAT-890"
+replaces: []
+superseded-by: []
+---
+
+# Disable Builder Service Account
+
+
+## Summary
+
+Provide cluster configuration options to disable the auto-creation of the
+`builder` service account. When this behavior is disabled, the `builder`
+service account and its associated RBAC should not be created in new
+namespaces, and cluster admins can delete `builder` service accounts in
+existing namespaces.
+
+
+## Motivation
+
+In OCP 4.14, `Build` and `DeploymentConfig` were added as optional install
+capabilities to OpenShift [1]. When `Build` and `DeploymentConfig` capabilities
+are not enabled, the APIs and respective controllers are not enabled on the
+cluster. Cluster admins can enable these capabilites after installation, but
+they cannot disable these capabilities once enabled.
+
+This feature will allow cluster administrators to disable the `builder` service
+account while keeping other components of the BuildConfig system available.
+When disabled, cluster administrators will be responsible for configuing a
+service account that can perform actions that typically occur during builds.
+Most notably, these service accounts will need permission to push to the
+OpenShift internal registry if that feature is enabled. The builder service
+account does not need permission to create pods with elevated pod security
+permissions, as this has been delegated to the build controller's service
+account.
+
+[1] https://issues.redhat.com/browse/WRKLDS-695
+
+### User Stories
+
+- As an enterprise platform engineer, I want a mechanism to disable the builder
+  service account - even if the “Build” capability is enabled on the cluster -
+  so that I can provide my own RBAC for builds in the “golden path” namespace
+  template for dev teams.
+- As an information security officer, I want to disable the builder service
+  account as part of our process to limit access to the OCP Build system on
+  production/application clusters so that only service accounts related to
+  applications are deployed, and they have the minimum permissions necessary.
+- As a software architect/platform engineer, I want to change the default
+  service account used for builds so I can customize its permissions.
+- As a product manager, I want to know how many OpenShift clusters are
+  disabling the builder service account so that I can understand the impact of
+  this feature.
+
+### Goals
+
+- On existing clusters that have the `Build` capability enabled, cluster admins
+  can disable the creation of the `builder` service account.
+- On existing clusters that the `builder` and service accounts are disabled,
+  cluster admins can enable creation of these service accounts in every
+  namespace.
+- When the `builder` service account is disabled, existing `builder` service
```

Comment:
if they delete them, manually, they won't be recreated correct? or if they modify their secrets, etc.